app/controllers/groups_controller.rb

```diff
@@ -0,0 +1,70 @@
+class GroupsController < ApplicationController
+  before_action :set_group, only: %i[show edit update destroy]
+
+  # GET /groups or /groups.json
+  def index
+    @groups = Group.all
+  end
+
+  # GET /groups/1 or /groups/1.json
+  def show
+    @group_entities = Entity.where(group_id: params[:id]).all
+  end
```

Comment:
- [ ]  You should avoid making SQL queries outside of models. It makes it harder to test and debug your app.